server/src/main/java/org/elasticsearch/index/fielddata/plain/SortedOrdinalsIndexFieldData.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.index.fielddata.plain;
+
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.index.OrdinalMap;
+import org.apache.lucene.index.SortedSetDocValues;
+import org.apache.lucene.search.SortField;
+import org.apache.lucene.search.SortedSetSortField;
+import org.elasticsearch.common.util.BigArrays;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.index.fielddata.IndexFieldData;
+import org.elasticsearch.index.fielddata.IndexFieldData.XFieldComparatorSource.Nested;
+import org.elasticsearch.index.fielddata.IndexFieldDataCache;
+import org.elasticsearch.index.fielddata.LeafOrdinalsFieldData;
+import org.elasticsearch.indices.breaker.CircuitBreakerService;
+import org.elasticsearch.script.field.ToScriptField;
+import org.elasticsearch.search.DocValueFormat;
+import org.elasticsearch.search.MultiValueMode;
+import org.elasticsearch.search.aggregations.support.ValuesSourceType;
+import org.elasticsearch.search.sort.BucketedSort;
+import org.elasticsearch.search.sort.SortOrder;
+
+import static org.elasticsearch.index.fielddata.IndexFieldData.XFieldComparatorSource.sortMissingLast;
+
+public class SortedOrdinalsIndexFieldData extends AbstractIndexOrdinalsFieldData {
+
+    public static class Builder implements IndexFieldData.Builder {
+        private final String name;
+        private final ToScriptField<SortedSetDocValues> toScriptField;
+        private final ValuesSourceType valuesSourceType;
+
+        public Builder(String name, ValuesSourceType valuesSourceType, ToScriptField<SortedSetDocValues> toScriptField) {
+            this.name = name;
+            this.toScriptField = toScriptField;
+            this.valuesSourceType = valuesSourceType;
+        }
+
+        @Override
+        public SortedOrdinalsIndexFieldData build(IndexFieldDataCache cache, CircuitBreakerService breakerService) {
+            return new SortedOrdinalsIndexFieldData(cache, name, valuesSourceType, breakerService, toScriptField);
+        }
+    }
+
+    public SortedOrdinalsIndexFieldData(
+        IndexFieldDataCache cache,
+        String fieldName,
+        ValuesSourceType valuesSourceType,
+        CircuitBreakerService breakerService,
+        ToScriptField<SortedSetDocValues> toScriptField
+    ) {
+        super(fieldName, valuesSourceType, cache, breakerService, toScriptField);
+    }
+
+    @Override
+    public SortField sortField(@Nullable Object missingValue, MultiValueMode sortMode, Nested nested, boolean reverse) {
+        SortField sortField = new SortField(getFieldName(), SortField.Type.STRING, reverse);
```

Comment:
I wonder if setting this to `SortField.Type.STRING` by default is allowed. 

If I leave it to the default `SortField.Type.CUSTOM` method `IndexSortConfig#validateIndexSortField` fails

Reply:
`Type.STRING` is correct here.

Reply:
`Type.CUSTOM` means that it's expecting you to give it a custom FieldComparator, which we don't need.  It is a horrible API from about 2004 and I keep meaning to hack it out and replace it with something less awful...